.github/workflows/coverage.yml

```diff
@@ -113,17 +113,31 @@ jobs:
           mkdir build && cd build;
 
           # CMake options & flags
-          export COVERAGE_FLAGS="--coverage -O0 -fno-inline -g";
-          export CMAKE_OPTION="-DNRN_ENABLE_MPI=ON -DNRN_ENABLE_INTERVIEWS=ON -DNRN_ENABLE_PYTHON=ON -DNRN_ENABLE_PYTHON_DYNAMIC=ON -DNRN_PYTHON_DYNAMIC=${PYTHON_MIN};${PYTHON_MAX} -DNRN_ENABLE_CORENEURON=ON -DNRN_ENABLE_PROFILING=ON -DNRN_ENABLE_BACKTRACE=ON -DNRN_ENABLE_MUSIC=ON -DCMAKE_PREFIX_PATH=${MUSIC_INSTALL_DIR} -DMUSIC_ROOT=${MUSIC_INSTALL_DIR}";
-          cmake $CMAKE_OPTION -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DNRN_ENABLE_TESTS=ON -DNRN_ENABLE_PERFORMANCE_TESTS=OFF -DCMAKE_C_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_CXX_FLAGS="${COVERAGE_FLAGS}" -DCORENRN_ENABLE_UNIT_TESTS=ON ..;
-
+          cmake_args=(-G Ninja \
+                      -DCMAKE_BUILD_TYPE=Debug \
+                      -DCMAKE_C_COMPILER="$CC" \
+                      -DCMAKE_CXX_COMPILER="$CXX" \
+                      -DNRN_ENABLE_BACKTRACE=ON \
+                      -DNRN_ENABLE_CORENEURON=ON \
```

Comment:
Looks like those tests are running: https://github.com/neuronsimulator/nrn/actions/runs/4719110674/jobs/8369510642#step:11:5432
And since the CoreNEURON -> NEURON merge then this variable is supposed to just be another name for NRN_ENABLE_TESTS.

Reply:
They are ON by default. Then my only guess would be switching to Ninja

Reply:
The CoreNEURON build flags are different too. This PR (https://github.com/neuronsimulator/nrn/actions/runs/4719110674/jobs/8369510642#step:11:171):
```
COMPILE FLAGS       |  -fopenmp  -g  -O0 -fstack-protector  -std=c++17 -DCORENEURON_BUILD -DHAVE_MALLOC_H -DNRNMPI=1 -DMPI_NO_CPPBIND=1 -DOMPI_SKIP_MPICXX=1 -DMPICH_SKIP_MPICXX=1 -DLAYOUT=0 -DDISABLE_HOC_EXP -DENABLE_SPLAYTREE_QUEUING -DCORENEURON_USE_LEGACY_UNITS=0 -DCORENEURON_CALIPER  -I/usr/local/include
```
vs. the reference (https://github.com/neuronsimulator/nrn/actions/runs/4698501847/jobs/8330858710#step:11:174):
```
COMPILE FLAGS       | --coverage -O0 -fno-inline -g -fopenmp  -g  -O2  -std=c++17 -DCORENEURON_BUILD -DHAVE_MALLOC_H -DNRNMPI=1 -DMPI_NO_CPPBIND=1 -DOMPI_SKIP_MPICXX=1 -DMPICH_SKIP_MPICXX=1 -DLAYOUT=0 -DDISABLE_HOC_EXP -DENABLE_SPLAYTREE_QUEUING -DCORENEURON_USE_LEGACY_UNITS=0 -DCORENEURON_CALIPER  -I/usr/local/include
```
partly because we didn't set `CMAKE_BUILD_TYPE=Debug` before. I **think** the lack of `--coverage` is a red herring...

Reply:
It looks like using the internal NEURON/CoreNEURON lists of flags instead of CMAKE_CXX_FLAGS has had a few side-effects:
- we no longer collect coverage for backward, interviews and mod2c (external/backward, external/iv and external/mod2c)
- we no longer collect coverage from CoreNEURON translated MOD files (`x86_64/corenrn/build/exp2syn.cpp` etc.)
- some RxD/Python coverage is no longer collected (src/nrnpython/inithoc.gcno, src/nrnpython/rxd_llgramarea.gcno, src/nrnpython/rxd_marching_cubes.gcno, share/lib/python/neuron/rxd/geometry3d/ctng.gcno, share/lib/python/neuron/rxd/geometry3d/graphicsPrimitives.gcno, share/lib/python/neuron/rxd/geometry3d/surfaces.gcno)
- some MUSIC coverage is no longer collected (src/neuronmusic/neuronmusic.gcno)

the first one seems fine/improvement, but the others should be fixed...

Reply:
Hopefully 21232dff30268c197f93b5a316b370cb49432c3d will address the last point.